### DISCUSSION TITLE
Automatic Updater

### DISCUSSION THREAD

**Discussion by H-Tevin** [Codectory/AutoActions#84] [Show and tell]
I'm not great at coding, but I made a batch file to automatically update AutoActions. You have to download the zip yourself, but this will extract it, exit AutoActions, copy the new files over the old ones, restart AutoActions, then delete the zip. It doesn't take a ton of time to do yourself, but why do work if you don't have to?

Here's the full code:
@echo off

cd C:\Program Files\7-Zip
taskkill /F /IM AutoActions.exe
7z e "C:\Users\"YOURNAME"\Downloads\Release_AutoActions_*.zip" -o"C:\Program Files\AutoActions" -y

del /f "C:\Users\"YOURNAME"\Downloads\Release_AutoActions_*.zip"
"C:\Program Files\AutoActions\AutoActions.exe"

exit

Just change the paths to be the same as yours and it should work. You also need 7 zip installed.
It'd be cool if something like this could be implemented officially. Since AutoActions has access to the internet to check for updates, it might also have the capability to download the files too.

Also, if anyone could figure out a way to automatically run this when a new github zip is downloaded, that would be great. Right now it has to be run manually.

Edit: I switched everything over to AutoActions with the branding change (a much-needed change by the way. The app can do so much more than HDR, it was time for a name change).

**Comment by Codectory** [comment]
I do have plans for impleminting a simple auto updater. Thanks for sharing you script!

**Comment by Todd491** [comment]
Thanks for your work so far!
Just signed up to github to +1 this suggestion. 
Also, to other people copying the above script, don't forget to rename AutoHDR to AutoActions...

**Reply by H-Tevin** [reply]
I switched everything over. Thankfully, the change was just in the name, so everything else is exactly the same. Thanks for reminding me to update it and I'm glad I could help!
